kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -72,324 +72,114 @@ public open class ApplicationCommandRegistry : KoinComponent {
+     */
+    public abstract suspend fun register(command: UserCommand<*>): UserCommand<*>?
 
-    /** Register multiple generic application commands. **/
-    public open suspend fun sync(
-        removeOthers: Boolean = false,
-        guildId: Snowflake?,
-        commands: List<ApplicationCommand<*>>
-    ) {
-        // NOTE: Someday, discord will make real i18n possible, we hope...
-        val locale = bot.settings.i18nBuilder.defaultLocale
+    /** Event handler for slash commands. **/
+    public abstract suspend fun handle(event: ChatInputCommandInteractionCreateEvent)
 
-        val guild = if (guildId != null) {
-            kord.getGuild(guildId)
-                ?: return logger.debug {
-                    "Cannot register application commands for guild ID ${guildId.asString}, " +
-                        "as it seems to be missing."
-                }
-        } else {
-            null
-        }
+    /** Event handler for message commands. **/
+    public abstract suspend fun handle(event: MessageCommandInteractionCreateEvent)
 
-        // Get guild commands if we're registering them (guild != null), otherwise get global commands
-        val registered = guild?.commands?.toList()
-            ?: kord.globalCommands.toList()
+    /** Event handler for user commands. **/
+    public abstract suspend fun handle(event: UserCommandInteractionCreateEvent)
 
-        if (!bot.settings.applicationCommandsBuilder.register) {
-            commands.forEach { commandObj ->
-                val existingCommand = registered.firstOrNull { commandObj.matches(locale, it) }
+    /** Unregister a slash command. **/
+    public abstract suspend fun unregister(command: SlashCommand<*, *>, delete: Boolean = true): SlashCommand<*, *>?
 
-                if (existingCommand != null) {
-                    when (commandObj) {
-                        is MessageCommand<*> -> messageCommands[existingCommand.id] = commandObj
-                        is SlashCommand<*, *> -> slashCommands[existingCommand.id] = commandObj
-                        is UserCommand<*> -> userCommands[existingCommand.id] = commandObj
-                    }
-                }
-            }
+    /** Unregister a message command. **/
+    public abstract suspend fun unregister(command: MessageCommand<*>, delete: Boolean = true): MessageCommand<*>?
 
-            return  // We're only syncing them up, there's no other API work to do
-        }
+    /** Unregister a user command. **/
+    public abstract suspend fun unregister(command: UserCommand<*>, delete: Boolean = true): UserCommand<*>?
 
-        // Extension commands that haven't been registered yet
-        val toAdd = commands.filter { c -> registered.all { !c.matches(locale, it) } }
+    // region: Utilities
 
-        // Extension commands that were previously registered
-        val toUpdate = commands.filter { c -> registered.any { c.matches(locale, it) } }
+    /** Unregister a message command. **/
```

Comment:
This doc comment is wrong.

Reply:
Heh? Why that?

Reply:
It's a generic unregistration function, not just for message commands.

Reply:
Oh lol. I looked at the wrong method, haha.